prometheus-server-metrics/src/main/scala/org/http4s/server/prometheus/PrometheusMetrics.scala

```diff
@@ -62,7 +62,8 @@ object PrometheusMetrics {
       serviceMetrics: ServiceMetrics,
       service: HttpService[F],
       emptyResponseHandler: Option[Status],
-      errorResponseHandler: Throwable => Option[Status]
+      errorResponseHandler: Throwable => Option[Status],
+      statusesToIgnore: Set[Status]
```

Comment:
Why just `Status`?  Might we want to ignore on headers? 

```scala
responseRule: Response[F] => Rule
```

where

```scala
sealed trait Rule
case object Record extends Rule
case object Reject extends Rule
```

And if we're filtering responses, I can also imagine filtering requests.

Reply:
@rossabaker I like this idea because we have in the past done some shuffling of which routes gets metrics so that we don't get our `ping` route in with business route metrics.  This would make it so I could just exclude `ping`.

Reply:
@rossabaker I'm finding this is getting complicated because there are spots where we create a status but we don't formally have a response, namely because of:
```
emptyResponseHandler: Option[Status],
errorResponseHandler: Throwable => Option[Status]
```

Reply:
@rossabaker I'm giving it a try though, will be done shortly.

Reply:
@rossabaker changed in 6b516b6 , let me know if anything needs cleaned up